src/widgets/calendar/lv_calendar_lunar.c

```diff
@@ -0,0 +1,261 @@
+    0x0D5252, 0x0DAA47, 0x66B53B, 0x056D4F, 0x04AE45, 0x4A4EB9, 0x0A4D4C, 0x0D1541, 0x2D92B5  /*2091-2099*/
+};
+
+static const uint16_t month_total_day[13] = {0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334, 365};
+
+
+static const char * lunar_month_name[] = {"正月", "二月", "三月", "四月", "五月", "六月",
+                                          "七月", "八月", "九月", "十月", "十一月", "腊月"
+                                         };
+
+static const char * lunar_day_name[] = {"初一", "初二", "初三", "初四", "初五",
+                                        "初六", "初七", "初八", "初九", "初十",
+                                        "十一", "十二", "十三", "十四", "十五",
+                                        "十六", "十七", "十八", "十九", "二十",
+                                        "廿一", "廿二", "廿三", "廿四", "廿五",
+                                        "廿六", "廿七", "廿八", "廿九", "三十"
+                                       };
+
+static const lv_calendar_festival_t lunar_festival[] = {
+    {"春节", 1, 1},
+    {"元宵节", 1, 15},
+    {"端午节", 5, 5},
+    {"七夕节", 7, 7},
+    {"中秋节", 8, 15},
+    {"重阳节", 9, 9},
+};
+
+static const lv_calendar_festival_t solar_festival[] = {
+    {"元旦", 1, 1},
+    {"情人节", 2, 14},
+    {"妇女节", 3, 8},
+    {"植树节", 3, 12},
+    {"消费节", 3, 15},
+    {"愚人节", 4, 1},
+    {"劳动节", 5, 1},
+    {"青年节", 5, 4},
+    {"儿童节", 6, 1},
+    {"建党节", 7, 1},
+    {"建军节", 8, 1},
+    {"教师节", 9, 10},
+    {"国庆节", 10, 1},
+    {"万圣节", 10, 31},
+    {"平安夜", 12, 24},
+    {"圣诞节", 12, 25},
+};
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+void lv_calendar_set_lunar_mode(lv_obj_t * obj, bool en)
+{
+    lv_calendar_t * calendar = (lv_calendar_t *)obj;
+    calendar->use_lunar_calendar = en;
+    lv_calendar_set_showed_date(obj, calendar->today.year, calendar->today.month);
+}
+
+const char * lv_calendar_get_day_name(lv_calendar_date_t * solar)
+{
+    uint16_t i, len;
+    lv_calendar_lunar_t lunar_date;
+    lunar_date = solar_to_lunar(solar);
+    if(solar->year > 2099 || solar->year < 1901)
+        return NULL;
+    len = sizeof(lunar_festival) / sizeof(lv_calendar_festival_t);
+    for(i = 0; i < len; i++) {
+        if((lunar_date.today.month == lunar_festival[i].month) &&
+           (lunar_date.today.day == lunar_festival[i].day) &&
+           lunar_date.leep_month == false)
+            return lunar_festival[i].festival_name;
+    }
+    len = sizeof(solar_festival) / sizeof(lv_calendar_festival_t);
+    for(i = 0; i < len; i++) {
+        if((solar->month == solar_festival[i].month) &&
+           (solar->day == solar_festival[i].day))
+            return solar_festival[i].festival_name;
+    }
+    if(lunar_date.today.day == 1) {
+        if(lunar_date.leep_month == false)
+            return lunar_month_name[lunar_date.today.month - 1];
+        else {
+            static char buff[15];
```

Comment:
```suggestion
            char buff[15];
```

no need static, alloc on stack is acceptable.

Reply:
maybe moving it to lv_calendar_t is more proper

Reply:
I think we should add another global array to handle.